pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -39,42 +42,43 @@ func (*AgentClusterInstall) Dependencies() []asset.Asset {
 
 // Generate generates the AgentClusterInstall manifest.
 func (a *AgentClusterInstall) Generate(dependencies asset.Parents) error {
-	// installConfig := &agent.OptionalInstallConfig{}
-	// dependencies.Get(installConfig)
-
-	// agentClusterInstall := &hiveext.AgentClusterInstall{
-	// 	ObjectMeta: metav1.ObjectMeta{
-	// 		Name:      "agent-cluster-install",
-	// 		Namespace: installConfig.Config.Namespace,
-	// 	},
-	// 	Spec: hiveext.AgentClusterInstallSpec{
-	// 		ClusterDeploymentRef: corev1.LocalObjectReference{
-	// 			Name: installConfig.Config.ObjectMeta.Name,
-	// 		},
-	// 		SSHPublicKey: installConfig.Config.SSHKey,
-	// 		ProvisionRequirements: hiveext.ProvisionRequirements{
-	// 			ControlPlaneAgents: int(*installConfig.Config.ControlPlane.Replicas),
-	// 		},
-	// 	},
-	// }
-
-	// var numberOfWorkers int = 0
-	// for _, compute := range installConfig.Config.Compute {
-	// 	numberOfWorkers = numberOfWorkers + int(*compute.Replicas)
-	// }
-	// agentClusterInstall.Spec.ProvisionRequirements.WorkerAgents = numberOfWorkers
-	// a.Config = agentClusterInstall
-
-	// agentClusterInstallData, err := yaml.Marshal(agentClusterInstall)
-	// if err != nil {
-	// 	return errors.Wrap(err, "failed to marshal agent installer AgentClusterInstall")
-	// }
-
-	// a.File = &asset.File{
-	// 	Filename: agentClusterInstallFilename,
-	// 	Data:     agentClusterInstallData,
-	// }
+	installConfig := &agent.OptionalInstallConfig{}
+	dependencies.Get(installConfig)
+
+	if installConfig.Config != nil {
+		agentClusterInstall := &hiveext.AgentClusterInstall{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "agent-cluster-install",
+				Namespace: installConfig.Config.Namespace,
+			},
+			Spec: hiveext.AgentClusterInstallSpec{
+				ClusterDeploymentRef: corev1.LocalObjectReference{
+					Name: installConfig.Config.ObjectMeta.Name,
+				},
+				SSHPublicKey: strings.Trim(installConfig.Config.SSHKey, "|\n\t"),
```

Comment:
Why this trimming is required? It doesn't seem in other locations of the installer where `installConfig.Config.SSHKey` is used

Reply:
I also trimmed it here for infra-env https://github.com/pawanpinjarkar/installer/blob/convert-install-config-to-ztp-assets/pkg/asset/agent/manifests/infraenv.go#L64

If we don't and if the user provided install-config has multiline in the `ssh-key` like [this](https://gist.github.com/pawanpinjarkar/ff1186f4dd5d9294612ea4cf4ccc27e3#file-install-config-yaml-L31-L32), for example, then the `create-cluster-and-infraenv.service` fails and the assisted-service pod throws below error

```
time="2022-07-20T15:27:02Z" level=error 
msg="Failed to register InfraEnv infraEnv with id 55231fc2-bcfc-4414-89d3-78fd19a6e9ea. 
Error: SSH key is not valid" 
func="github.com/openshift/assisted-service/internal/bminventory.(*bareMetalInventory)
.RegisterInfraEnvInternal.func1" 
file="/src/internal/bminventory/inventory.go:3891" 
cluster_id=55231fc2-bcfc-4414-89d3-78fd19a6e9ea go-id=35
pkg=Inventory request_id=11d830d2-f13c-4d40-9ddb-6942466ee3ee
```